tests/test_core.py

```diff
@@ -986,6 +987,32 @@ def test_partial_method_lots_of_args(self):
         )
         self.assertEqual(buf.value.decode("utf-8"), pystring)
 
+    def test_objcmethod_objcpartialmethod_objcboundmethod_repr(self):
+        """Test ObjCMethod, ObjCPartialMethod, and ObjCBoundMethod repr"""
+
+        obj = NSObject.new()
+        bound_method = obj.init
+        method = bound_method.method
+        receiver = bound_method.receiver
+        partial_method = obj.performSelector.method
+
+        # Check ObjCBoundMethod repr
+        objcboundmethod_repr = repr(obj.init)
+        self.assertIn(ObjCBoundMethod.__qualname__, objcboundmethod_repr)
+        self.assertIn(repr(method), objcboundmethod_repr)
+        self.assertIn(str(receiver), objcboundmethod_repr)
```

Comment:
Tests usually benefit from being explicit, even if there's content that *can* be obtained programmatically. In this case, the name of the bound method is technically provided by `__qualname__`, but it's better to have an *explicit* test of a *literal* string, rather than relying on the implementation of `__qualname__` to do the right thing. 

Adding to that - a `assertRegex` would be a better match for these tests. Using `assertIn` allows any output ordering, as long as it includes the right parts; a regex-based test lets you check the exact output, with an explicit carveout for the characters that will vary based on memory address.